lib/route-recognizer.ts

```diff
@@ -426,63 +417,56 @@ class RouteRecognizer {
     normalizeSegment, normalizePath, encodePathSegment
   };
 
-  constructor() {
-    this.rootState = new State();
-    this.names = {};
-  }
-
   add(routes: Route[], options?: { as: string }) {
     let currentState = this.rootState;
-    let regex = "^";
-    let types = { statics: 0, dynamics: 0, stars: 0 };
-    let handlers: any[] = new Array(routes.length);
+    let pattern = "^";
+    let types: [number, number, number] = [0, 0, 0];
+    let handlers: Handler[] = new Array(routes.length);
     let allSegments: Segment[] = [];
     let name: string | undefined;
 
     let isEmpty = true;
-
+    let j = 0;
```

Comment:
because it is only supposed to iterate over the new ones, the issue is we used to concat the results to the all but loop over just the new results, I'll add a comment.